cmd/utils/auth.go

```diff
@@ -32,3 +49,82 @@ func GetCredentials() (*azidentity.ChainedTokenCredential, error) {
 
 	return chain, nil
 }
+
+// cachedInteractiveBrowserCredential is a credential that uses the interactive browser to authenticate and caches the token.
+// TODO: This is a workaround until the azidentity package supports caching, https://github.com/Azure/azure-sdk-for-go/issues/16643.
+type cachedInteractiveBrowserCredential struct {
+	client public.Client
+}
+
+func newCachedInteractiveBrowserCredential() (*cachedInteractiveBrowserCredential, error) {
+	file := path.Join(homeDir(), ".kubectl-az", "token-cache.json")
+	if err := os.MkdirAll(path.Dir(file), 0700); err != nil {
+		return nil, fmt.Errorf("creating cache directory: %w", err)
+	}
+
+	client, err := public.New(developerSignOnClientID,
+		public.WithCache(&tokenCache{file: file}),
+		public.WithAuthority(runtime.JoinPaths(string(azidentity.AzurePublicCloud), organizationsTenantID)),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("creating public client: %w", err)
+	}
+
+	return &cachedInteractiveBrowserCredential{client: client}, nil
+}
+
+// GetToken implements the azcore.TokenCredential interface on cachedInteractiveBrowserCredential.
+func (c *cachedInteractiveBrowserCredential) GetToken(ctx context.Context, options policy.TokenRequestOptions) (*azcore.AccessToken, error) {
+	// TODO: may be this can be improved with https://github.com/Azure/kubectl-az/issues/11
```

Comment:
Yep, agree we should allow users to select the account to use. Maybe the config file is a good place.